src/sampleworks/models/protocol.py

```
@@ -0,0 +1,233 @@
+from jaxtyping import Float
+
+from sampleworks.utils.framework_utils import Array
+
+
+C = TypeVar("C")
+StructureModelOutputT = TypeVar("StructureModelOutputT", covariant=True)
+FlowOrEnergyBasedModelOutputT = TypeVar("FlowOrEnergyBasedModelOutputT")
+
+
+@dataclass
+class GenerativeModelInput(Generic[C]):  # noqa: UP046 (for Python 3.11 compatibility)
```

Ruff (0.14.14)

[warning] Unused `noqa` directive (non-enabled: `UP046`)

Remove unused `noqa` directive

(RUF100)